src/gtk/window.cpp

```diff
@@ -37,16 +37,15 @@
 
 #include <ctype.h>
 
-#include <gtk/gtk.h>
-#include <gdk/gdkkeysyms.h>
 #include "wx/gtk/private.h"
-#include "wx/gtk/private/gtk2-compat.h"
 #include "wx/gtk/private/gtk3-compat.h"
 #include "wx/gtk/private/event.h"
 #include "wx/gtk/private/win_gtk.h"
 #include "wx/private/textmeasure.h"
 using namespace wxGTKImpl;
 
+#include <gdk/gdkkeysyms.h>
```

Comment:
This will cause a compile error with GTK+ < 2.20. `gdkkeysyms.h` needs to come before `gtk2-compat.h`.

Reply:
Are you sure about this? AFAICS we don't really use the constants from `gdkkeysyms.h` in `wx/gtk/private/gtk2-compat.h`, we just do `#define GDK_KEY_XXX GDK_XXX`. And it only matters that `GDK_XXX` are defined before they're used in `src/gtk/window.cpp`.

Or am I missing something here?

Reply:
```
../src/gtk/window.cpp: In function ‘long int wxTranslateKeySymToWXKey(KeySym, bool)’:
../src/gtk/window.cpp:564:14: error: ‘GDK_KEY_Shift_L’ was not declared in this scope
         case GDK_KEY_Shift_L:
              ^~~~~~~~~~~~~~~
```
`gtk2-compat.h` won't define the new names unless the old names are already defined (i.e. `gdkkeysyms.h` has already been included.

Reply:
> `gtk2-compat.h` won't define the new names unless the old names are already defined

Ah, this was what I was missing, thanks.

The simplest way to fix this would be to just always include `gdk/gdkkeysyms.h` from `gtk2-compat.h`. Do you see any problems with this? Looking at the GDK header, it's a pretty trivial one, so there shouldn't be any risk nor overhead from including it from there.